config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -169,6 +209,43 @@ private Optional<ConfigDocument> getLatestVersionConfigDoc(
     return Optional.empty();
   }
 
+  private Map<ConfigResourceContext, Optional<ConfigDocument>> getLatestVersionConfigDocs(
+      Set<ConfigResourceContext> configResourceContexts) throws IOException {
+    // build filter
+    Filter[] childFilters = new Filter[configResourceContexts.size()];
```

Comment:
If we're forced to use an array here (why did we use arrays in an API written this decade?!), we can still use a list and switch it over at the last second with toArray() so we don't ahve to deal with indices.